wrangler-core/src/main/java/io/cdap/wrangler/utils/ColumnConverter.java

```diff
@@ -284,4 +292,17 @@ private static Object convertType(String col, String toType, Object object) thro
         String.format("Column '%s' has value of type '%s' and cannot be converted to a '%s'.", col,
             object.getClass().getSimpleName(), toType));
   }
+
+  private static BigDecimal setDecimalScale(BigDecimal decimal, Integer scale, RoundingMode roundingMode)
+    throws DirectiveExecutionException {
+    if (scale == null) {
+      return decimal;
+    }
+    try {
+      return decimal.setScale(scale, roundingMode);
+    } catch (ArithmeticException e) {
+      throw new DirectiveExecutionException(
+        String.format("'rounding-mode' cannot be '%s' for the specified scaling operation", roundingMode));
```

Comment:
Add `e` as a cause.  Also I don't think the message is telling the story. The most often case would be using RoundingMode.UNNESSESARY and the rounding is needed. It's better to say something about the value and scale tried, e.g. that scale can't be set to X for value Y with rounding mode Z

Reply:
Done, added a better error message